include/caf/sensor_manager.h

```diff
@@ -14,14 +14,19 @@ extern "C" {
 #include <zephyr/device.h>
 #include <zephyr/drivers/sensor.h>
 
+#define FLOAT_TO_SENSOR_VALUE(float_val)					\
+	{									\
+		.val1 = (int32_t)float_val,					\
+		.val2 = (int32_t)(float_val - (int32_t)float_val) * 1000000.0,	\
```

Comment:
since val1 and val2 are int, implicit cast would be fine I expect as it would allow second argument to be converted back to float and then subtraction and multiplication would happen on floats

The interesting thing is precision :)

as for int32_max we need to follow what was chosen in sensor value type in zephyr

Reply:
maybe we should allow user to define sensor value explicitly to avoid loosing precision?

Reply:
Don't understand why there is a problem with value that is val2 multiplicated by.

From sensor_value:

Representation of a sensor readout value.
 *
 * The value is represented as having an integer and a fractional part,
 * and can **be obtained using the formula val1 + val2 * 10^(-6).** 

So we have to multiply by 1000000.0 or otherwise we would have a wrong value later.